src/test/java/LoginStatusServletTest.java

```diff
@@ -0,0 +1,125 @@
+import java.util.HashMap;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public final class LoginStatusServletTest {
+  private static final String DOMAIN_NAME = "gmail.com";
+  private static final String USER_EMAIL = "test@gmail.com";
+  private static final String USER_ID = "testID";
+
+  private static final String HOME_PAGE_URL = "/";
+  private static final String LOGIN_PAGE_URL = "/login.html";
+
+  private static final String LOGIN_URL = "/_ah/login?continue\u003d%2F";
+  private static final String LOGOUT_URL = "/_ah/logout?continue\u003d%2Flogin.html";
+
+  private static final AuthenticationInformation LOGGED_OUT_INFORMATION =
+      new AuthenticationInformation(LOGIN_URL);
+  private static final AuthenticationInformation LOGGED_IN_INFORMATION =
+      new AuthenticationInformation(LOGOUT_URL, USER_EMAIL);
+
+  // Uses local UserService.
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalUserServiceTestConfig())
+          .setEnvEmail(USER_EMAIL)
+          .setEnvAuthDomain(DOMAIN_NAME)
+          .setEnvAttributes(new HashMap(
+              ImmutableMap.of("com.google.appengine.api.users.UserService.user_id_key", USER_ID)));
+
+  @Mock private HttpServletRequest request;
+  @Mock private HttpServletResponse response;
+
+  private LoginStatusServlet servlet;
+  private UserService userService;
+
+  @Before
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    userService = UserServiceFactory.getUserService();
+
+    servlet = new LoginStatusServlet(userService);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void doGetIfUserIsNotLoggedIn() throws IOException {
+    helper.setEnvIsLoggedIn(false);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+    writer.flush();
+
+    String actual = getJsonResponse(stringWriter);
+    String expected = new Gson().toJson(LOGGED_OUT_INFORMATION);
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void doGetIfUserIsLoggedIn() throws IOException {
+    helper.setEnvIsLoggedIn(true);
+
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    when(response.getWriter()).thenReturn(writer);
+
+    servlet.doGet(request, response);
+    writer.flush();
+
+    String actual = getJsonResponse(stringWriter);
+    String expected = new Gson().toJson(LOGGED_IN_INFORMATION);
+    Assert.assertEquals(expected, actual);
```

Comment:
Since you are constructing the expected string and testing the actual response, I think it would be best to only manipulate the expected string. In this case, you could just add a newline character to the end of it.